### PR TITLE
[msbuild][tests] Clarify FrameworkListTest failure message

### PR DESCRIPTION
**Pull request by VincentDondain**

### CODE REVIEW

**Comment by VincentDondain** [comment]
@akoeplinger your feedback is welcome as you got the test failure and it might not have been totally clear (:

File: msbuild/tests/Xamarin.iOS.Tasks.Tests/FrameworkListTest.cs [hunk 1]
```diff
@@ -29,12 +29,12 @@ public void CheckFrameworkListFile (string frameworkListFile)
 
 			foreach (var assembly in frameworkListAssemblies) {
 				if (!installedAssemblies.Any (a => a.Name == assembly.Name))
-					ReportAssemblies (assembly, $"Incorrectly added assemblies in '{frameworkListFile}'");
+					ReportAssemblies (assembly, $"Incorrectly listed assemblies in '{frameworkListFile}'. Check if those assemblies/facades exist in the final SDK root folder.");
```
Comment (akoeplinger) [inline msbuild/tests/Xamarin.iOS.Tasks.Tests/FrameworkListTest.cs]: I'd reword this as:
```
One or more assemblies listed in '{frameworkListFile}' were not found in the final SDK root folder. Update the list if an assembly was intentionally removed.
```

It makes it clearer what the expectation of the test is.

File: msbuild/tests/Xamarin.iOS.Tasks.Tests/FrameworkListTest.cs [hunk 2]
```diff
@@ -29,12 +29,12 @@ public void CheckFrameworkListFile (string frameworkListFile)
...
 			}
 
 			foreach (var assembly in installedAssemblies) {
 				if (!frameworkListAssemblies.Any (a => a.Name == assembly.Name))
-					ReportAssemblies (assembly, $"Missing assemblies in '{frameworkListFile}'");
+					ReportAssemblies (assembly, $"Missing assemblies in '{frameworkListFile}'. We have detected assemblies/facades in the SDK root folder that are not listed in the 'FrameworkList' file.");
```
Comment (akoeplinger) [inline msbuild/tests/Xamarin.iOS.Tasks.Tests/FrameworkListTest.cs]: This is already quite good, maybe just align it with the above message if you go with my suggestion:
```
One or more assemblies in the the SDK root folder are not listed in '{frameworkListFile}'.  Update the list if an assembly was intentionally added.
```

**Comment by mandel-macaque** [comment]
+1 on @akoeplinger sentence.

**Comment by mandel-macaque** [comment]
build

**Comment by VincentDondain** [comment]
build
